pkgs/applications/version-management/mercurial/default.nix

```diff
@@ -1,6 +1,6 @@
-{ stdenv, fetchurl, python, makeWrapper, docutils, unzip, hg-git, dulwich
-, guiSupport ? false, tk ? null, curses, cacert
-
+{ stdenv, fetchurl, python, makeWrapper, docutils, unzip, curses, cacert
+, guiSupport ? false, tk ? null
+, gitSupport ? false, hg-git ? null, dulwich ? null
 , ApplicationServices }:
 
```

Comment:
It's common to add an assert here to ensure that `hg-git` and `dulwich` are not `null` when `gitSupport` is enabled.